src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/transformers/ThreadTransformer.kt

```diff
@@ -57,6 +66,40 @@ internal class ThreadTransformer(
         visitInsn(opcode)
     }
 
+    override fun visitMaxs(maxStack: Int, maxLocals: Int) = adapter.run {
+        if (isThreadRunMethod(methodName, desc)) {
+            visitLabel(runMethodTryBlockEnd)
+            visitLabel(runMethodCatchBlock)
+
+            // STACK: exception
+            dup()
+            invokeStatic(Injections::onThreadException)
+            // STACK: exception, isSuppressed
+
+            // Notify that the thread has finished.
+            // TODO: currently does not work, because `ManagedStrategy::onThreadFinish`
```

Comment:
Is there any plan to fix this? What the absence of the `invokeStatic(Injections::afterThreadFinish)` call might lead to?

Reply:
> Is there any plan to fix this?

Yes, I am going to fix it later in another PR, as it is not critical. I've created an issue for now: https://github.com/JetBrains/lincheck/issues/530

> What the absence of the invokeStatic(Injections::afterThreadFinish) call might lead to?

Apparently, there are no serious problems because of that. The only issue is that after termination the thread will end up in `Aborted` state instead of `Finished` state, which leads to few inconveniences in the code.